base/linalg/eigen.jl

```diff
@@ -155,7 +155,9 @@ function eig(A::AbstractMatrix, B::AbstractMatrix)
     F.values, F.vectors
 end
 function eig(A::Number, B::Number)
-    F = eigfact(A,B)
+    a = fill(A,1,1)
+    b = fill(B,1,1)
+    F = eigfact(a,b)
```

Comment:
I don't think so. `eigfact(A::Number, B::Number)` is a pretty rare edge case, one we can deal with in fewer lines by just putting the `Number`s in arrays.